### ISSUE TITLE
Use of this code yields "Below" lines to stdout

### ISSUE THREAD

**Issue by joonas-fi** [oylenshpeegul/epochs#1]
I traced it to here:

https://github.com/oylenshpeegul/epochs/blob/8ef2270514bf86684af1cadb84bf3918c325d66a/epochs.go#L35

Looks like a forgotten debug statement? Should this be an error?

**Comment by oylenshpeegul** [comment]
Thanks for your interest! You're right, that is more a debug statement. I am doing the arithmetic with [big.Float](https://golang.org/pkg/math/big/#Float.Int64) and then converting to an int64. I wonder if I should simply return the (int64, Accuracy) pair, instead of checking the Accuracy like that. I have not encountered this overflow case. How were you calling it?

**Comment by joonas-fi** [comment]
I'm using your library for NTFS epochs. IIRC this is reproducible with
simply trying to convert time.Now() into NTFS time.

If needed, I can provide some sample code tomorrow

**Comment by oylenshpeegul** [comment]
I see. Right you are. Running this code

```go
	now := time.Now()
	fmt.Println(now)

	ntfs := epochs.ToWindowsFile(now)
	fmt.Println(ntfs)

	fmt.Println(epochs.WindowsFile(ntfs))
```

yields

```
$ go run .
2021-03-22 17:27:22.46376231 -0400 EDT m=+0.000113097
Below
132609220424637623
2021-03-22 21:27:22.4637623 +0000 UTC
```

I think we should just remove that Println you originally pointed out and return the int64 without comment. If you think that Accuracy would be useful, let me know and I will think about refactoring.

**Comment by joonas-fi** [comment]
It took me a while to understand what the `math.Big` accuracy means. Now I understand after writing some tests:

```go
native := time.Date(2020, 3, 23, 10, 17, 0, 123456750, time.UTC)

// going from native timestamp to NTFS time rounds to 100-nanosecond accuracy:
//    123456750
// => 123456700
//           ^ !!
nativeRounded := TimeFrom(native).Parse()

assert.EqualJson(t, native.Format(time.RFC3339Nano), `"2020-03-23T10:17:00.12345675Z"`)
assert.EqualJson(t, nativeRounded.Format(time.RFC3339Nano), `"2020-03-23T10:17:00.1234567Z"`)

TimeFrom(native) // this yields another "Below"
TimeFrom(nativeRounded) // this doesn't yield "Below" because big.Float's internal representation is integer b/c it doesn't exceed NTFS time accuracy
```

(you can read my `TimeFrom` and `Time` as `ntfs.TimeFrom(time.Time)` and `ntfs.Time.Parse()time.Time`)

Basically this [math/big.Float.Int64 docs](https://pkg.go.dev/math/big#Float.Int64):

> if x is an integer

The internal representation of `big.Float` is not an integer when given time is more accurate than 100-nanoseconds :)

To your question:

> If you think that Accuracy would be useful, let me know and I will think about refactoring.

In context of NTFS timestamps, this rounding is by design (NTFS times, like all times, have limited accuracy), uninteresting and harmless (thus it shouldn't be an error or information return).

But another question, related to this accuracy check, is the Int64() accuracy documentation:

> The result is (math.MinInt64, Above) for x < math.MinInt64, and (math.MaxInt64, Below) for x > math.MaxInt64

These conditions would not just result in loss of accuracy, but completely wrong calculation result, so I think it should be accounted for as an error. How that error should be handled, panic or error return for all APIs.. I'm not sure, neither sounds great. But I'd think `panic()` is at least better than ignoring the error entirely.

Food for thought :)

**Comment by oylenshpeegul** [comment]
Okay, I made it panic for the latter cases and ignore Accuracy for the former case. At least, for now. I'll have to think about whether returning errors everywhere would be better.

Thanks for you help!

Commit 2141bb22b8dd9dd04e9c5b15cda0582a805ab36d closes issue #1

**Comment by joonas-fi** [comment]
Good work, and thanks for responding quickly!

And thanks for making this library! I had sample code for parsing `time.Time` from NTFS time, but I was struggling to go in the other direction (`time.Time` to NTFS time) due to calculation overflowing, I didn't even think `math/big` could help me LOL. I'm glad you made this library and that I found it!
